chain/network/src/store/schema/mod.rs

```diff
@@ -0,0 +1,279 @@
+            KnownPeerStatus::Connected => primitives::KnownPeerStatus::Connected,
+            KnownPeerStatus::Banned(r, t) => primitives::KnownPeerStatus::Banned(
+                r,
+                t,
+            ),
+        }
+    }
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct KnownPeerStateRepr {
+    peer_info: primitives::PeerInfo,
+    status: KnownPeerStatus,
+    /// UNIX timestamps in nanos.
+    first_seen: u64,
+    last_seen: u64,
+}
+
+impl BorshRepr for KnownPeerStateRepr {
+    type T = primitives::KnownPeerState;
+    fn to_repr(s: &primitives::KnownPeerState) -> Self {
+        Self {
+            peer_info: s.peer_info.clone(),
+            status: s.status.clone().into(),
+            first_seen: s.first_seen,
+            last_seen: s.last_seen,
+        }
+    }
+
+    fn from_repr(s: Self) -> Result<primitives::KnownPeerState, Error> {
+        Ok(primitives::KnownPeerState {
+            peer_info: s.peer_info,
+            status: s.status.into(),
+            first_seen: s.first_seen,
+            last_seen: s.last_seen,
+        })
+    }
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct EdgeRepr {
+    key: (PeerId, PeerId),
+    nonce: u64,
+    signature0: Signature,
+    signature1: Signature,
+    removal_info: Option<(bool, Signature)>,
+}
+
+impl BorshRepr for EdgeRepr {
+    type T = primitives::Edge;
+
+    fn to_repr(e: &Self::T) -> Self {
+        Self {
+            key: e.key().clone(),
+            nonce: e.nonce(),
+            signature0: e.signature0().clone(),
+            signature1: e.signature1().clone(),
+            removal_info: e.removal_info().cloned(),
+        }
+    }
+    fn from_repr(e: Self) -> Result<Self::T, Error> {
+        Ok(primitives::Edge::new(e.key.0, e.key.1, e.nonce, e.signature0, e.signature1)
+            .with_removal_info(e.removal_info))
+    }
+}
+
+/////////////////////////////////////////////
+// Columns
+
+pub struct AccountAnnouncements;
+impl Column for AccountAnnouncements {
+    const COL: DBCol = DBCol::AccountAnnouncements;
+    type Key = AccountIdFormat;
+    type Value = Borsh<AnnounceAccount>;
+}
+
+pub struct Peers;
+impl Column for Peers {
+    const COL: DBCol = DBCol::Peers;
+    type Key = Borsh<PeerId>;
+    type Value = KnownPeerStateRepr;
+}
+
+pub struct PeerComponent;
+impl Column for PeerComponent {
+    const COL: DBCol = DBCol::PeerComponent;
+    type Key = Borsh<PeerId>;
+    type Value = Borsh<u64>;
+}
+
+pub struct ComponentEdges;
+impl Column for ComponentEdges {
+    const COL: DBCol = DBCol::ComponentEdges;
+    type Key = U64LE;
+    type Value = Vec<EdgeRepr>;
+}
+
+pub struct LastComponentNonce;
+impl Column for LastComponentNonce {
+    const COL: DBCol = DBCol::LastComponentNonce;
+    type Key = Borsh<()>;
+    type Value = Borsh<u64>;
+}
+
+////////////////////////////////////////////////////
+// Storage
+
+// Parsing error.
+#[derive(Error, Debug)]
```

Comment:
```suggestion
#[derive(thiserror::Error, Debug)]
```

https://github.com/near/nearcore/blob/master/docs/style.md#derives

Reply:
obsolete